activation/activation.go

```diff
@@ -381,24 +424,21 @@ func (b *Builder) buildNIPostChallenge(ctx context.Context) (*types.NIPostChalle
 	case <-b.syncer.RegisterForATXSynced():
 	}
 	current := b.currentEpoch()
-	prev, err := b.cdb.GetLastAtx(b.nodeID)
-	if err != nil {
-		if !errors.Is(err, sql.ErrNotFound) {
-			return nil, err
-		}
-	} else if prev.PublishEpoch == current+1 {
-		current += 1
+	prev, err := b.cdb.GetLastAtx(b.signer.NodeID())
+	switch {
+	case err == nil:
+		current = max(current, prev.PublishEpoch)
```

Comment:
TBH, I don't understand why the current epoch is overwritten here. What if the previous ATX was published a year ago etc.? The `current++` part below won't fix this.

Reply:
If the previous ATX was published a year ago it's publish epoch will be 26 less then `current` -> `max(current, current-26)` is `current`.

Before the code was:

```go
if prev.PublishEpoch == current+1 {
	current += 1
}
```

there seems to have been a need for this check. It also makes intuitively sense; if there was a previous ATX we want to set `current` to at least its `PublishEpoch` so we don't accidentally try to publish two ATXs in the same epoch. But the check itself was weird: why only check if publish epoch was `current+1`?

Regarding the `current++` check I added below: I added this to avoid the situation where if one misses the registration window for the poet round the node would complain about it every 5 minutes from PoET round start (Monday morning) until the end of the epoch (Friday morning). It also prevents clearing the challenge in the case this happens with the initial ATX.